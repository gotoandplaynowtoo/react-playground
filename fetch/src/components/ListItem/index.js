import styled from 'styled-components';

const ListItem = styled.div`
  padding: 4px;
  margin: 4px 0;
  background: #eee;
  color: #222;
`;

export default ListItem;