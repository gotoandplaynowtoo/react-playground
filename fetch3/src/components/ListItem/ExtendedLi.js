import Li from 'components/Li';

const ExtendedLi = Li.extend`
  background: #eee;
  margin: 4px 0;
  padding: 4px;
`;

export default ExtendedLi;