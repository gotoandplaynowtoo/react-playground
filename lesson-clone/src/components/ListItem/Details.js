import styled from 'styled-components';

const Details = styled.div`
  display: flex;
  padding: 4px;
`;

export default Details;