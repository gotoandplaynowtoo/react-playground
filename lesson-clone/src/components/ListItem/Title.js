import styled from 'styled-components';

const Title = styled.div`
  font-weight: bold;
  padding: 4px;
  text-transform: uppercase;
`;

export default Title;