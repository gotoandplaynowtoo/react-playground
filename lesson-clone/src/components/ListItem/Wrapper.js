import styled from 'styled-components';

const Wrapper = styled.div`
  background: #eee;
  display: flex;
  justify-content: space-between;  
  padding: 4px;
`;

export default Wrapper;