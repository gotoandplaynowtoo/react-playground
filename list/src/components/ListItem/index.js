import styled  from 'styled-components';

const ListItem = styled.li`
  padding: 4px;
`;

export default ListItem;