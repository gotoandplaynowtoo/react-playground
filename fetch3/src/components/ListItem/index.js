import React from 'react';
import ExtendedLi from './ExtendedLi';

const ListItem = (props) => {
  return <ExtendedLi>{props.content}</ExtendedLi>
};

export default ListItem;