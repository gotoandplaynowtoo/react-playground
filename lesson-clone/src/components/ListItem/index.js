import React from 'react';
import PropTypes from 'prop-types';
import Wrapper from './Wrapper';
import Title from './Title';
import Details from './Details';
import Button from 'components/Button';

const ListItem = (props) => {
  const {
    objectID,
    title,
    url,
    author,
    points,
    onClick
  } = props;

  return (
    <Wrapper>
      <div>
        <Title><a href={ url }>{ title }</a></Title>
        <Details>
          <div>author: {author}</div>
          <div>|</div>
          <div>points: {points}</div>
        </Details>
      </div>
      <div>
        <Button value={objectID} onClick={onClick}>Dismiss</Button>
      </div>
    </Wrapper>
  );
};

ListItem.propTypes = {
  onClick: PropTypes.func
};

export default ListItem;