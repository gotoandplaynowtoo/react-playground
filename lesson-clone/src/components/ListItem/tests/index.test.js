import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ListItem from 'components/ListItem';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<ListItem/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('it renders correctly', () => {
  const tree = renderer.create(<ListItem/>).toJSON();
  expect(tree).toMatchSnapshot();
});