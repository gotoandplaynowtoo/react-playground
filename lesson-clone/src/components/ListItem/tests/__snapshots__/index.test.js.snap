// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders correctly 1`] = `
<div
  className="sc-bdVaJa fNLcSE"
>
  <div>
    <div
      className="sc-bwzfXH wTuGn"
    >
      <a
        href={undefined}
      />
    </div>
    <div
      className="sc-htpNat cmYzTD"
    >
      <div>
        author: 
      </div>
      <div>
        |
      </div>
      <div>
        points: 
      </div>
    </div>
  </div>
  <div>
    <button
      className="sc-bxivhb fdMNis"
      onClick={undefined}
      value={undefined}
    >
      Dismiss
    </button>
  </div>
</div>
`;
